docs: split building and deploying doc

skoh7645 opened the pull request:
<!--
Please read the [Code of Conduct](https://github.com/appsody/website/blob/master/CODE_OF_CONDUCT.md) and the [Contributing Guidelines](https://github.com/appsody/website/blob/master/CONTRIBUTING.md) before opening a pull request.
-->

## Checklist
<!-- For completed items, change [ ] to [x]. -->

- [x] commit message follows [commit guidelines](https://github.com/appsody/website/blob/master/CONTRIBUTING.md#commit-message-guidelines).
- [x] website builds and runs in production - for information on how to test locally go [here](https://github.com/appsody/website/blob/master/DEVELOPMENT.md).

## Description
Split up the "building and deploying" doc into " building", "deploying" and "appsody operator" docs. I tried to keep the content largely the same in this PR. Future PRs will be made to improve the actual content. 

For easier reviewing:
- [ ] "Building" doc is taken from the original "building and deploying" doc. I only added the first 2 paragraphs and the last line
- [ ] "Deploying" doc is the original "building and deploying" doc with the "building" and "appsody operator" sections removed from it. I only made changes to the `# Getting ready to deploy your Appsody project` section, and the last sentence in the `### Deployment via the Appsody Operator` section
- [ ] "Appsody Operator" doc is directly taken from the original "building and deploying" doc with no additional changes

## Related Issues
 Related to https://github.com/appsody/website/issues/359

helenmasters commented on content/docs/reference/appsody-operator.md:
```diff
@@ -0,0 +1,122 @@
+---
+
+# How the Appsody operator works
+Before we delve into the details of `appsody deploy` and `appsody operator`, we need to spend a few words on how the Appsody operator works.
```
This seems like a strange place to start a page.... "Before we delve into...?" I think given this is now on its own page you can lose this whole sentence.

helenmasters commented:
You'll need to add redirects, changes to the sidebar yaml and any direct website refs also need updating...

helenmasters commented on content/docs/using-appsody/building.md:
```diff
@@ -0,0 +1,37 @@
+
+The Appsody CLI downloads and runs a *development* Docker image of the target runtime during local development. This image differs slightly from the image that is used at deployment time, because it configures tools that are useful only during the development phase.
+
+The Appsody CLI provides the [`appsody build`](/docs/cli-commands/#appsody-build) command to generate a *deployment* Docker image on your local Docker registry. 
```
It feels like this should be the first sentence and the sentence/paragraph above should be a note/comment underneath, as it seems strange to talk about development first on a page about build.

skoh7645 replied:
I've tried to reword it taking this into account :)

helenmasters commented on content/docs/using-appsody/deploying.md:
```diff
@@ -0,0 +1,112 @@
+
+The Appsody CLI provides the `appsody deploy` command to build and deploy a Docker image directly to a Kubernetes cluster that you are using for testing or staging.
+
+You can delegate the build and deployment steps to an external pipeline, such as a Tekton pipeline that consumes the source code of your Appsody project after you push it to a GitHub repository. Within the pipeline, you can run `appsody build`, which builds the application image and generates a deployment manifest. You can use the manifest to deploy your application to a Kubernetes environment wh ...
```
You should probably link back to the build page where you reference `appsody build`.

skoh7645 commented:
I have added one redirect which is the main one. I think we can add redirects for anchor links that we are aware have been used externally. However, we will probably have to accept some breakages as we can't redirect every anchor header. I've also updated the sidebar yaml and any  direct website link references

helenmasters commented:
Sandy, I dreamt about the docs last night (sad I know!). Is there a link from the deploying doc to the operator docs as I didn't notice one yesterday? There should be something to show users that there is more information in Reference otherwise we can't expect users to find it.

skoh7645 commented:
@helenmasters haha sorry to hear about that. There is a link to the operator doc at the end of the `### Deployment via the Appsody Operator` section
